internal/pkg/agent/install/install_windows.go

```diff
@@ -55,12 +63,23 @@ func fixInstallMarkerPermissions(markerFilePath string, ownership utils.FileOwne
 }
 
 // withServiceOptions just sets the user/group for the service.
-func withServiceOptions(username string, groupName string) ([]serviceOpt, error) {
+func withServiceOptions(username string, groupName string, password string) ([]serviceOpt, error) {
 	if username == "" {
 		// not installed with --unprivileged; nothing to do
 		return []serviceOpt{}, nil
 	}
 
+	if password != "" {
+		if isFullDomainName, err := isWindowsDomainUsername(username); err != nil {
+			return nil, fmt.Errorf("failed to parse username: %w", err)
+		} else if !isFullDomainName {
+			return nil, fmt.Errorf("username is not in proper format 'domain\\username'")
```

Comment:
This error message could be clearer, one of the failure reasons is an illegal character and that isn't listed here.